django/core/management/commands/inspectdb.py

```diff
@@ -101,16 +101,18 @@ def table2model(table_name):
                     column_name = row.name
                     is_relation = column_name in relations
 
+                    is_pk = column_name == primary_key_column
+                    has_pk = primary_key_column
                     att_name, params, notes = self.normalize_col_name(
-                        column_name, used_column_names, is_relation)
+                        column_name, used_column_names, is_relation, is_pk, has_pk)
                     extra_params.update(params)
                     comment_notes.extend(notes)
 
                     used_column_names.append(att_name)
                     column_to_field_name[column_name] = att_name
 
                     # Add primary_key and unique, if necessary.
-                    if column_name == primary_key_column:
+                    if is_pk or (column_name == 'id' and not has_pk):
```

Comment:
I don't think that we should treat `ID` columns as primary keys if we don't find any other primary key, it's just too much guessing IMO. It's documented that:

_"This feature is meant as a shortcut, **not as definitive model generation**. After
you run it, you'll want to look over the generated models yourself to make
customizations."_